src/cli_util.rs

```diff
@@ -2279,8 +2279,14 @@ fn handle_early_args(
     args: &[String],
     layered_configs: &mut LayeredConfigs,
 ) -> Result<(), CommandError> {
-    // ignore_errors() bypasses errors like "--help" or missing subcommand
-    let early_matches = app.clone().ignore_errors(true).get_matches_from(args);
```

Comment:
`get_matches_from()` -> `try_get_matches_from()?` makes sense - we should try to propagate the error.